python/frontend/cirrus/cirrus/lambdas/feature_hashing_helper.py

```diff
@@ -0,0 +1,31 @@
+""" Helper functions for feature hashing """
+
+import hashlib
+import json
+import struct
+
+import boto3
+
+
+def hash_data(data, columns, N):
+    """ Replace the appropriate columns for this row. """
+    col_set = set([int(i) for i in columns])
+    for idx, row in enumerate(data):
```

Comment:
I don't think this works if we hash a subset of the columns. In that case we will mix the non-hashed columns with the hashed columns.

Reply:
Hmm my interpretation was that those kind of collisions are just a part of how feature hashing works. How should I handle it differently? In `InputReader.cpp`, it looks like we're either hashing [all of the features] or excluding the [non-categorical features].

[all of the features]: https://github.com/jcarreira/cirrus/blob/master/src/InputReader.cpp#L892
[non-categorical features]: https://github.com/jcarreira/cirrus/blob/master/src/InputReader.cpp#L592

Reply:
Yes, the C++ was done for a single case.

The way I would do this is to move all columns that are not hashed to the range ```[0, num_of_columns_not_hashed-1]```. All the other columns go to the range ```[num_of_columns_not_hashed, num_of_columns_not_hashed+N]```.

Reply:
Since I'm not sure how many columns there are in the data without traversing through it, I opted to shift the columns not hashed to the end of the range instead (`[0, num_columns_hashed]` and the original columns are shifted by `num_columns_hashed`)